### PR TITLE
NetBSD build fixes

### PR DESCRIPTION
**Pull request by 0-wiz-0**
These patches make xmoji build on NetBSD.

### CODE REVIEW

File: src/bin/xmoji/filewatcher.c [hunk 1]
```diff
@@ -1,4 +1,6 @@
+#ifdef __linux__
```
Comment (Zirias) [inline src/bin/xmoji/filewatcher.c:1]: Maybe just define the glibc-specific _DEFAULT_SOURCE instead? IMHO better than checking for Linux, glibc might be used on a different OS and Linux might be used with a different libc...

The effect should be the same, _DEFAULT_SOURCE causes glibc to just "expose everything", and we'd rely on other libc implementations to do the same when *no* feature test macro is defined...

File: src/bin/xmoji/unistr.h [hunk 2]
```diff
@@ -4,7 +4,11 @@
 #include "macros.h"
 
 #include <poser/decl.h>
-#include <uchar.h>
+#include <stddef.h>
```
Comment (Zirias) [inline src/bin/xmoji/unistr.h:7]: I'll try to move this to a separate compatibility header that still uses `uchar.h` when available.

Comment (Zirias) [reply src/bin/xmoji/unistr.h:7]: Done in commit [a28946b](https://github.com/Zirias/xmoji/commit/a28946b23e013f3c596b801a5c6f1a75c2000c90), could you please double-check?

Comment (0-wiz-0) [reply src/bin/xmoji/unistr.h:7]: You missed a spot :)
tools/emojigen/src/emojigen.c uses uchar.h too.

File: src/bin/xmoji/filewatcher.c (hunk 1 above)
Comment (0-wiz-0) [reply src/bin/xmoji/filewatcher.c:1]: Using `_DEFAULT_SOURCE` is fine with me, that symbol doesn't mean anything on NetBSD.

File: src/bin/xmoji/unistr.h (hunk 2 above)
Comment (Zirias) [reply src/bin/xmoji/unistr.h:7]: thanks! Will fix and then add to master. I'll also pull the statvfs.

**Comment by Zirias** [comment]
Individual commits pushed to master, thank you!
Closing here ...

**Pull request closed without merging**